src/main/java/it/pagopa/transactions/utils/TransactionsUtils.java

```diff
@@ -10,21 +10,79 @@
 import org.springframework.stereotype.Component;
 import reactor.core.publisher.Mono;
 
+import java.util.EnumMap;
+import java.util.Map;
+import java.util.Set;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+
 @Component
 public class TransactionsUtils {
 
     private final TransactionsEventStoreRepository<Object> eventStoreRepository;
 
+    private static final Map<it.pagopa.ecommerce.commons.generated.server.model.TransactionStatusDto, it.pagopa.generated.transactions.server.model.TransactionStatusDto> transactionStatusLookupMap = new EnumMap<>(
+            it.pagopa.ecommerce.commons.generated.server.model.TransactionStatusDto.class
+    );
+
     @Autowired
     public TransactionsUtils(TransactionsEventStoreRepository<Object> eventStoreRepository) {
         this.eventStoreRepository = eventStoreRepository;
     }
 
+    static {
+        Set<String> commonsStatuses = Set
+                .of(it.pagopa.ecommerce.commons.generated.server.model.TransactionStatusDto.values())
+                .stream()
+                .map(Enum::toString)
+                .collect(Collectors.toSet());
+        Set<String> transactionsStatuses = Set
+                .of(it.pagopa.generated.transactions.server.model.TransactionStatusDto.values())
+                .stream()
+                .map(Enum::toString)
+                .collect(Collectors.toSet());
+        /*
+         * @formatter:off
+         * In case of statuses enumeration mismatch an `IllegalArgumentException` is thrown, preventing the module from being
+         * run and avoiding runtime errors correlated to specs updated into commons
+         * that are not reflected into this service specs (such as a transaction status added only into commons).
+         * The above exception message reports the detected statuses enumeration mismatch between commons and transactions values.
+         * @formatter:on
```

Comment:
```suggestion
         * @formatter:off
         *
         * In case of statuses enumeration mismatch an `IllegalArgumentException` is thrown, preventing the module from being
         * run and avoiding runtime errors correlated to specs updated into commons
         * that are not reflected into this service specs (such as a transaction status added only into commons).
         * The above exception message reports the detected statuses enumeration mismatch between commons and transactions values.
         *
         * @formatter:on
```